tez-api/src/main/java/org/apache/tez/common/TezUtils.java

```diff
@@ -51,6 +57,14 @@
 public final class TezUtils {
 
   private static final Logger LOG = LoggerFactory.getLogger(TezUtils.class);
+  private static final int PROPERTY_THRESHOLD;
+  private static final boolean PROPERTY_MASK;
+
+  static {
+    TezConfiguration c = new TezConfiguration();
```

Comment:
I'm afraid this approach will become confusing hard to handle over time
a new TezConfiguration() in a static initializer will always pick up whatever is present in tez-site.xml, hence we don't really have control over that: I think if we want to be able to use the size threshold, we need to achieve that from the actual config, which might need a bit of refactoring as this method receives Configuration as an iterable (and we need to see the threshold and mask property beforehand without iterating over the values)

Reply:
I reworked the PR (https://github.com/apache/tez/pull/308/commits/17e07e7c714d670cfbd5e15eb3c56f33a022f2ec,https://github.com/apache/tez/pull/308/commits/687bd3b22f26288c0f49d40d05588d4f17a7b2fa) to allow using the size threshold from within the configuration object.